n_utils/nameless-dt-print-aws-profiles.cpp

```diff
@@ -1,51 +1,49 @@
+#include <filesystem>
 #include <fstream>
 #include <iostream>
 #include <string>
-#include <unordered_set>
+#include <set>
+#include <string_view>
 
-int main(int argc, const char* argv[]) {
-    std::string prefix = "";
-    std::unordered_set<std::string> ret;
-    if (argc > 1) {
-        prefix = argv[1];
-    }
-    if (!getenv("HOME")) {
+/// Print all unique AWS profile names found from config.
+int main(const int argc, const char* argv[]) {
+    const auto home = std::getenv("HOME");
+    if (!home) {
         return 0;
     }
-    std::string home = getenv("HOME");
-    std::ifstream credentials(home + "/.aws/credentials", std::fstream::in);
-    if (credentials) {
-        std::string line;
-        while (getline(credentials, line)) {
-            if (line[0] != '[' || line[line.length() - 1] != ']') {
-                continue;
-            }
-            std::string profile_name = line.substr(1, line.length() - 2);
-            if (strncmp(profile_name.c_str(), prefix.c_str(), prefix.size()) == 0) {
-                ret.insert(profile_name);
-            }
-        }
-        credentials.close();
+
+    std::string_view prefix;
```

Comment:
std::string and string_view does not need to be assigned with `""`, the default constructor will create an empty string automatically